### PR TITLE
Update & Deploy Kubespray

### PR DESCRIPTION
**Pull request by HassanAbouelela**
This PR moves the kubespray inventory, updates parts of it, and documents deployment steps.

There are a lot of files, particularly in 232f631, which have been moved up from the kubespray directory. Because that directory was a submodule, there is no real diff, however I've made minimal changes which are as follows:
* Documented and linked docs for `upstream_dns_servers` in `group_vars/all.yml`
* Added the credentials folder to the gitignroe
* Combined the inventory file at the project root with the one used for kubespray
* Added a README
* Removed a dead download URL for calicoctl in `roles/downloads/defaults/main.yml`

I also made the submodule an https URL instead of an SSH one - not everyone has SSH configured, especially if they're working on WSL because they have to run ansible :P

These changes have already been deployed to netcup prod. Testing pending.

### CODE REVIEW

File: inventory/inventory.ini [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# node4
+# node5
+# node6
+
+[calico_rr]
+
+[k8s_cluster:children]
+kube_control_plane
+kube_node
+calico_rr
```
Comment (jchristgit) [inline inventory/inventory.ini:38]: Can we remove this file? I don't understand what it's doing at the moment. Artifact of kubespray?

**Review by jchristgit — APPROVED** [review]
Rock solid man. You didn't just take the ball and drove it home, you took the entire stadium with you as well. Awesome work.

File: inventory/inventory.ini (hunk 1 above)
Comment (HassanAbouelela) [reply inventory/inventory.ini:38]: It’s generated when running in the init script, not sure if it’s actually used anywhere.

Comment (HassanAbouelela) [reply inventory/inventory.ini:38]: Not part of the init script, just the sample inventory https://github.com/kubernetes-sigs/kubespray/blob/master/inventory/sample/inventory.ini

**Pull request merged**